### ISSUE TITLE
RPC BC Break: `replace_file_source` => `update_file_source`

### ISSUE THREAD

**Issue by dantleech** [phpactor/phpactor#550]
This change will shortly be in `develop`.

**TL;DR**: Add an `update_file_source` method which delegates to `replace_file_source` in the plugin.

The RPC methods which update source code were using `replace_file_source`. They now (or will shortly) use `update_file_source`.

The old `replace_file_source` is structured as follows:

```
{
    "path": "/path/to/file.php"
    "source": "<?php the raw source code
}
```

The new `update_file_source` looks like this:

```
{
    "path": "/path/to/file.php"
    "source": "<?php the raw source code"
    "edits": [
        {
            "start": { "line": 1, "character": 1 },
            "length": 0,
            "text": "text to insert",
        },
        {
            // ...
        }
    ]
}
```

The `source` is included currently to simply allow the `update_file_source` to behave as the previous version did, but should be considered deprecated.

The text edits are a all the changes that need to be made to the document. A text edit:

- `start` position: line number and character (not byte!) offset
- `length`: number of characters to replace at the position
- `text`: the text to insert

This allows documents to be updated more efficiently and for the cursor position to be maintained.
